internal/tui/tui.go

```diff
@@ -280,6 +291,74 @@ func (a appModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 		}
 		return a, nil
 
+	case dialog.CloseSetupDialogMsg:
+		a.showSetupDialog = false
+
+		err := config.Update(func(cfg *config.Config) {
+			// Add Agent
+			if cfg.Agents == nil {
+				cfg.Agents = make(map[config.AgentName]config.Agent)
+			}
+			cfg.Agents[config.AgentPrimary] = config.Agent{
+				Model:     msg.Model.ID,
+				MaxTokens: msg.Model.DefaultMaxTokens,
+			}
+			cfg.Agents[config.AgentTitle] = config.Agent{
+				Model:     msg.Model.ID,
+				MaxTokens: 80,
+			}
+			cfg.Agents[config.AgentTask] = config.Agent{
+				Model:     msg.Model.ID,
+				MaxTokens: msg.Model.DefaultMaxTokens,
+			}
+
+			// Add Provider
+			if cfg.Providers == nil {
+				cfg.Providers = make(map[models.ModelProvider]config.Provider)
+			}
+
+			cfg.Providers[msg.Provider] = config.Provider{
+				APIKey: msg.APIKey,
+			}
+
+			cfg.SetupComplete = true
+		})
+		if err != nil {
+			slog.Debug("Failed to update config", "error", err)
+			panic(err)
+		}
+
+		// Reinitialize the agent
+		a.app.PrimaryAgent, err = agent.NewAgent(
+			config.AgentPrimary,
+			a.app.Sessions,
+			a.app.Messages,
+			agent.PrimaryAgentTools(
+				a.app.Permissions,
+				a.app.Sessions,
+				a.app.Messages,
+				a.app.History,
+				a.app.LSPClients,
+			),
+		)
+		if err != nil {
+			slog.Debug("Failed to initialize agent", "error", err)
+			panic(err)
+		}
+
+		// Show init dialog if project is not initialized
+		shouldShowInit, err := config.ShouldShowInitDialog()
+		if err != nil {
+			status.Error("Failed to check init status: " + err.Error())
+			return a, nil
+		}
+		if shouldShowInit {
+			a.showInitDialog = true
+			return a, nil
+		}
```

Comment:
my vote is that we just automatically run the project init command as well, don't hit them with a double modal situation.

Reply:
Can we address this in a follow up PR? The scope of this PR is already fairly large.

I will follow up afterwards since that means removing these dialogs.